.github/workflows/cargo.yml

```diff
@@ -21,19 +22,18 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: ${{ matrix.rust }}
-          override: true
-      - uses: Swatinem/rust-cache@v1
-      - uses: actions-rs/cargo@v1
+      - uses: actions/checkout@v4
         with:
-          command: build
-      - uses: actions-rs/cargo@v1
+          persist-credentials: false
+      - uses: dtolnay/rust-toolchain@master
+        id: toolchain
         with:
-          command: test
+          toolchain: ${{ matrix.rust }}
+      - uses: Swatinem/rust-cache@v2
+      - name: Build 1
+        run: cargo +${{steps.toolchain.outputs.name}} build --all-features
```

Comment:
I think the action sets the version as default so you could technically get rid of this (unless you want to be explicit)

Reply:
Thanks!